test/compilable/shared.d

```diff
@@ -0,0 +1,109 @@
+/* REQUIRED_ARGS: -preview=nosharedaccess
+TEST_OUTPUT:
+---
+pure nothrow @nogc ref @safe shared(C1)(return ref shared(C1) c)
+pure nothrow @nogc ref @safe shared(int)(return ref shared(C3) c)
+---
+*/
+ref shared(int) f(return shared ref int y)
```

Comment:
Fwiw, `return ref shared int` feels proper to me, because it could also be written `return ref shared(int)`.
It would probably be good if D required that storage classes appeared before type qualifiers, but alas, we should lead by example.

Reply:
could also be written as `return ref shared scope int`

Reply:
I would probably move `scope` left, because it's not a type qualifier (but it should be!)
We should standardise an order of appearance for the overwhelming sea of attributes in D these days.